src/librustc/middle/reachable.rs

```diff
@@ -37,21 +37,30 @@ use hir::intravisit;
 
 // Returns true if the given set of generics implies that the item it's
 // associated with must be inlined.
-fn generics_require_inlining(generics: &hir::Generics) -> bool {
-    generics.params.iter().any(|param| param.is_type_param())
+fn generics_require_inlining(generics: &ty::Generics) -> bool {
```

Comment:
Doesn't this usecase already have a method on `ty::Generics`?

Reply:
I don't think so.

Reply:
I meant the behavior, not name. I'm pretty sure it's `requires_monomorphization`.

Reply:
Okay, I'll change it.